modules/organization-policy/README.md

```diff
@@ -0,0 +1,166 @@
+# Google Cloud Organization Policy
+
+This module allows creation and management of [GCP Organization Policies](https://cloud.google.com/resource-manager/docs/organization-policy/org-policy-constraints) by defining them in a well formatted `yaml` files or with HCL.
+
+Yaml based factory can simplify centralized management of Org Policies for a DevSecOps team by providing a simple way to define/structure policies and exclusions.
+
+## Example
+
+### Terraform code
+
+```hcl
+# using configuration provided in a set of yaml files
+module "org-policy-factory" {
+  source = "./modules/organization-policy"
+
+  config_directory = "./policies"
+}
+
+# using configuration provided in the module variable
+module "org-policy" {
+  source = "./modules/organization-policy"
+  
+  organization_policies = {
+    "folders/1234567890" = {
+        "constraints/iam.disableServiceAccountKeyUpload" = {
+          rules = [
+            {
+                enforce = true
+            }
+          ]
+        }
+    },
+    "organizations/1234567890" = {
+      "run.allowedIngress" = {
+        rules = [
+          {
+            condition = {
+              description= "allow ingress"
+              expression = "resource.matchTag('123456789/environment', 'prod')"
+              title = "allow-for-prod-org"
+            },
+            values = {
+              allowed_values = ["internal"]
+            }
+          }
+        ]
+      }
+    }
+  } 
+}
+# tftest skip
+```
+
+## Org Policy definition format and structure
+
+### Structure of `organization_policies` variable
+
+```hcl
+organization_policies = {
+  "parent_id" = { # parent id in format projects/project-id, folders/1234567890 or organizations/1234567890.
+    "policy_name" = { # policy constraint id, for example compute.vmExternalIpAccess.
+      inherit_from_parent = true|false # (Optional) Only for list constraints. Determines the inheritance behavior for this policy.
+      reset               = true|false # (Optional) Ignores policies set above this resource and restores the constraint_default enforcement behavior.
+      rules               = [ # Up to 10 PolicyRules are allowed.
+        {
+          allow_all = true|false # (Optional) Only for list constraints. Setting this to true means that all values are allowed.
+          deny_all  = true|false # (Optional) Only for list constraints. Setting this to true means that all values are denied.
```

Comment:
I'd probably try to include `allow_all` and `deny_all` under the `values` object. Or perhaps even explode `values` inside the rule object.

Reply:
Yep, I would try and simplify the interface (see my other comment above).

Reply:
I'm trying to simply keep it consistent with the API itself, so may just check the API reference or the resource itself](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/org_policy_policy#argument-reference) and understand how to fill up the var. But if you think we may simplify the interface, I'm open to options.